src/ruby/Grammar.rb

```diff
@@ -38,7 +40,7 @@ def self.print_failure(text)
     puts '===================================='
     text.each_line.with_index { |line, i| puts "#{i + 1} #{line}" }
     puts "====================================\n\n"
-    puts parser.failure_reason
+    puts @parser.failure_reason
```

Comment:
Because `@parser` is the variable, `parser` is undefined. It wasn't printing anything before